server/host_controls_api.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import (
+	"github.com/gorilla/mux"
+	"net/http"
+)
+
+func (p *Plugin) handleMakeHost(w http.ResponseWriter, r *http.Request) {
+	var res httpResponse
+	defer p.httpAudit("handleMakeHost", &res, w, r)
+
+	userID := r.Header.Get("Mattermost-User-Id")
+	callID := mux.Vars(r)["call_id"]
+	newHostID := mux.Vars(r)["new_host_id"]
```

Comment:
Maybe we should chat more in 1:1 but I'm not sure how it's not really REST anymore... I would think the fact that it's a POST and I'm using url params is worse :). This could be a GET, except for the symantic that we're changing something, so maybe the reason to put it in JSON is simply because that's what you're "supposed" to do for POSTs? I dunno, the REST conventions are so arbitrary I can't keep track of it all. tbh, it's just simpler to put the params in the url here. 😝

Reply:
Apologies, I mistyped before. I meant to write `extensible` because having the payload makes it way easier to make non-breaking changes in the future. Agreed the URL looks cleaner this way though so not a big deal.

Reply:
Ahah, yes, that makes sense. :)